From 64adf4c7a0cd9ab0e49be71b8cbf2214394262d5 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Wed, 6 Jun 2018 10:33:59 +0800
Subject: [PATCH 3937/5242] MLK-18535-11 ARM64: dts: imx8mm-evk: enable
 'lcdif', 'mipi_dsi' and 'adv7535'

commit  16f288e00fe0892e2429b55dc479a2860fffbd81 from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable the whole display pipeline 'LCDIF --> MIPI DSI --> ADV7535'
on IMX8MM EVK board.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts |   28 ++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
index d3ca42e..e02a964 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
@@ -597,6 +597,20 @@
 	pinctrl-0 = <&pinctrl_i2c2>;
 	status = "okay";
 
+	adv_bridge: adv7535@3d {
+		compatible = "adi,adv7533";
+		reg = <0x3d>;
+		adi,addr-cec = <0x3b>;
+		adi,dsi-lanes = <4>;
+		status = "okay";
+
+		port {
+			adv7535_from_dsim: endpoint {
+				remote-endpoint = <&dsim_to_adv7535>;
+			};
+		};
+	};
+
 	typec1_ptn5110: tcpci@50 {
 		compatible = "usb,tcpci";
 		pinctrl-names = "default";
@@ -715,6 +729,20 @@
 	};
 };
 
+&lcdif {
+	status = "okay";
+};
+
+&mipi_dsi {
+	status = "okay";
+
+	port@1 {
+		dsim_to_adv7535: endpoint {
+			remote-endpoint = <&adv7535_from_dsim>;
+		};
+	};
+};
+
 &mu {
 	status = "okay";
 };
-- 
1.7.9.5

